### PR TITLE
Refactor singleton usage and make it possible to use with engines

### PR DESCRIPTION
**Pull request by palkan**
Replaces https://github.com/rails/webpacker/pull/1808.

Added docs on how to use Webpacker from Rails engine.

### CODE REVIEW

File: lib/webpacker/helper.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 module Webpacker::Helper
+  # Returns current Webpacker instance.
+  # Could be overriden to use multiple Webpacker
+  # configurations within the same app (e.g. with engines)
+  def current_webpacker
```
[the remark this answers is not in the document]
Comment (palkan) [reply lib/webpacker/helper.rb]: Agree.

[the remark this answers is not in the document]
Comment (palkan) [reply lib/webpacker/helper.rb]: Done

**Comment by michalgritzbach** [comment]
Any plans on merging this?

**Comment by gauravtiwari** [comment]
Thanks @palkan 🙏 sorry it took me a while to merge

**Comment by lazylester** [comment]
@palkan Thanks for this, esp the docs. In the docs you link to the Rails Guide section on Rails Engines, and you mention that the engines guide includes how to add webpacker to a rails engine. However the linked guide docs don't mention webpacker. Is there an unmerged PR somewhere for the Engines Guide that discusses adding webpacker?

**Pull request merged**